src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -211,6 +211,20 @@ public override BoundNode VisitCompoundAssignmentOperator(BoundCompoundAssignmen
             }
         }
 
+        public override BoundNode Visit(BoundNode node)
+        {
+            if (_inExpressionLambda &&
+                // Ignoring BoundConversion nodes prevents redundant diagnostics
+                !(node is BoundConversion) &&
+                node is BoundExpression expr &&
```

Comment:
I think it's still easier to override `Visit(BoundExpression)` instead of `BoundNode`

Reply:
I tried changing `Visit(BoundNode node)` to `VisitExpressionWithNullability(BoundExpressionWithNullability node)` and it never gets called :/